src/base1/Makefile.am

```diff
@@ -23,6 +23,8 @@ basedebug_DATA = \
 	dist/base1/cockpit.min.js.map \
 	$(NULL)
 
+AM_TESTS_ENVIRONMENT = export G_DEBUG=fatal-criticals ;
```

Comment:
I would suggest fatal-warnings too ... we use ```g_message()``` for expected failures. Both ```g_warning()``` and ```g_critical()``` (or related precondition warnings) are indicative of a test failure.